### PR TITLE
Addition of encoding parameters

### PR DESCRIPTION
**Pull request by lauerik**
Additional encoding parameters deemed applicable for parameter optimization added to encoder and made available as command line arguments.

### CODE REVIEW

File: src/opendlv-video-h264-encoder.cpp [hunk 1]
```diff
@@ -83,31 +137,63 @@ int32_t main(int32_t argc, char **argv) {
...
+                parameters.sSpatialLayers[0].iMaxSpatialBitrate = I_BITRATE_MAX;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceMode = SliceModeEnum::SM_SIZELIMITED_SLICE;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceNum = 1;
+
+                /*
+                 * Thesis parameters
+                 * https://github.com/cisco/openh264/wiki/TypesAndStructures
+                 * https://github.com/cisco/openh264/blob/master/codec/encoder/core/inc/param_svc.h#L132
+                 */
+                if (I_NUM_REF_FRAME==0){parameters.iNumRefFrame = AUTO_REF_PIC_COUNT;}
```
Comment (chrberger) [inline src/opendlv-video-h264-encoder.cpp]: Please add a blank and a new line around the `{` brackets for readability.

File: src/opendlv-video-h264-encoder.cpp [hunk 2]
```diff
@@ -83,31 +137,63 @@ int32_t main(int32_t argc, char **argv) {
...
+                else {parameters.iNumRefFrame = I_NUM_REF_FRAME;}
```
Comment (chrberger) [inline src/opendlv-video-h264-encoder.cpp]: Same here.

File: src/opendlv-video-h264-encoder.cpp [hunk 3]
```diff
@@ -83,31 +137,63 @@ int32_t main(int32_t argc, char **argv) {
...
+                parameters.bEnableLongTermReference = B_LONG_TERM_REFERENCE;
+                parameters.iLoopFilterDisableIdc = I_LOOP_FILTER;
+                parameters.bEnableDenoise = B_DENOISE;
+                parameters.bEnableBackgroundDetection = B_BACKGROUND_DETECTION;
+                parameters.bEnableAdaptiveQuant = B_ADAPTIVE_QUANT;
+                parameters.bEnableFrameCroppingFlag = B_FRAME_CROPPING;
+                parameters.bEnableSceneChangeDetect = B_SCENE_CHANGE_DETECT;
+
+                switch (RC_MODE) {
+                    case 0: {parameters.iRCMode = RC_MODES::RC_QUALITY_MODE; break;}
```
Comment (chrberger) [inline src/opendlv-video-h264-encoder.cpp]: Please add a blank after `{` and before `}`

**Review by chrberger — CHANGES_REQUESTED** [review]
Some minor cosmetic changes.

**Comment by guslauer** [comment]
> Some minor cosmetic changes.

File: src/opendlv-video-h264-encoder.cpp (hunk 1 above)
Comment (lauerik) [reply src/opendlv-video-h264-encoder.cpp]: Fixed in 7a978cd

File: src/opendlv-video-h264-encoder.cpp (hunk 2 above)
Comment (lauerik) [reply src/opendlv-video-h264-encoder.cpp]: Fixed in 7a978cd

File: src/opendlv-video-h264-encoder.cpp (hunk 3 above)
Comment (lauerik) [reply src/opendlv-video-h264-encoder.cpp]: Fixed in 7a978cd

**Review by chrberger — APPROVED** [review]

**Comment by chrberger** [comment]
Your PR is merged:
* openh264-encoder: https://github.com/chalmers-revere/opendlv-video-h264-encoder/releases/tag/v0.0.2
* frame-feed-evaluator: https://github.com/se-research/video-codec-performance-for-autonomous-driving/commit/8e50015f24d8666221cbca78098adcb9f216d5a8

**Pull request merged**